### PR TITLE
add collaborators

### PR DESCRIPTION
**Pull request by tatianahutii**

### CODE REVIEW

**Comment by Potapy4** [comment]
@tatianahutii please, fix merge conflicts, thanks!

File: backend/IDE.BLL/Services/RightsService.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            _context = context;
+        }
+
+        public async Task<ProjectRightsDTO> GetUserRightsForProject(int projectId, int userId)
+        {
+            var project = await _context.Projects.FirstOrDefaultAsync(p => p.Id == projectId);
+            if (project.AuthorId == userId)
+                return new ProjectRightsDTO() { IsAuthor = true };
+            
+            var projectMember = _context.ProjectMembers.FirstOrDefault(pm => pm.UserId == userId && pm.ProjectId == projectId);
```
Comment (muroslavko) [inline backend/IDE.BLL/Services/RightsService.cs]: make this method async

File: frontend/src/app/modules/collaborator/components/add-collaborators-list/add-collaborators-list.component.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    // public change() {
+    //     this.onChanged.emit(this.collaborators);
+    // }
+
+    public getUserAccess(user: CollaboratorDTO): SelectItem {
+        switch (user.access) {
+            case 0:
+                this.label = "Can read"
```
Comment (muroslavko) [inline frontend/src/app/modules/collaborator/components/add-collaborators-list/add-collaborators-list.component.ts]: Why don't you use this.userAccess?

File: frontend/src/app/modules/workspace/workspace-root/workspace-root.component.html [hunk 3]
```diff
@@ -1,9 +1,11 @@
 <div class="top-panel p-grid p-nogutter">
   <div class="hor-navbar p-col-12">
 
-    <button pButton type="button" icon="pi pi-play" iconPos="left" class="ui-button-secondary" label="Run"></button>
-    <button pButton type="button" icon="pi pi-arrow-down" iconPos="left" class="ui-button-secondary" label="Build"></button>
+    <button pButton *ngIf="IsBigger(3) || isAuthor()" type="button" icon="pi pi-play" iconPos="left" class="ui-button-secondary" label="Run"></button>
```
Comment (muroslavko) [inline frontend/src/app/modules/workspace/workspace-root/workspace-root.component.html]: `IsBigger(3)` what does it mean?

File: backend/IDE.BLL/Services/RightsService.cs [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+            }
+            else
+            {
+                throw new NonAuthorRightsChange();
+            }
+        }
+
+        public async Task SetRightsToProject(UpdateUserRightDTO update, int userId)
+        {
+            var project = _context.Projects.FirstOrDefault(p => p.Id == update.ProjectId);
```
Comment (muroslavko) [inline backend/IDE.BLL/Services/RightsService.cs]: use async methods

**Pull request merged**